### ISSUE TITLE
Support src= and <source> elements in UI library

### ISSUE THREAD

**Issue by joeyparrish** [google/shaka-player#2088]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes

**Is your feature request related to a problem? Please describe.**
It would be nice to be able to use something similar to the declarative `src=` and `<source>` elements used in HTML5 video.

**Describe the solution you'd like**
```html
<video data-shaka-player autoplay poster="foo.jpg" src="foo.mpd"></video>

<video data-shaka-player autoplay poster="foo.jpg"> 
  <source src="foo.mpd">  <!-- try this first -->
  <source src="foo.m3u8"> <!-- fall back to this if first fails to load -->
</video>
```

**Describe alternatives you've considered**
Keep the status quo, and use `player.load()` programattically.

**Comment by ismena** [comment]
@joeyparrish would you expect this declaration to trigger auto load and play?

**Comment by joeyparrish** [comment]
Auto-load would be triggered by `<source>`.  Auto-play would still be controlled by the `<video>` element's `autoplay` attribute.

**Comment by joeyparrish** [comment]
This was actually released in the v2.5 branch more than a year ago, and the issue never got closed.
